### DISCUSSION TITLE
Issue with sharing to EinkBro from "Palabre" Android application

### DISCUSSION THREAD

**Discussion by CottonEvoker** [plateaukao/einkbro#300] [Q&A] [answered]
Hi all,

I'm aware the discussion title begins "Issue", but I'm opting to submit in Discussions rather than Issues, as I suspect that what I describe may not be an EinkBro-side issue but a problem of some kind with the other app involved, Palabre.

First up: EinkBro is a remarkable achievement: Superb fullscreen toggle, and the gesture customizability is truly wonderful. Well done to all involved with it and in its develpoment.

Currently using eInkBro only on various stardard LCD-screen Android devices (OnePlus phone & Lenovo tablet): I wanted to familiarise myself with it ahead of using it on the Android-powered eInk device I intend acquiring soon. 

The Problem: When I share a link (e.g. to a news story) to EinkBro using the standard Android share menu from the Android news feed application Palabre, instead of the link opening in EinkBro, the webpage that opens in EinkBro is a search in DuckDuckGo of the url, requiring inconveniencingly that I then tap on the result in the DDG search results to view the story.

Notes:

* Palabre is a sort of Feedly alternative whose layout & funtionality I prefer to Feedly's. Unlike Feedly however, which is updated regularly, Palabre hasn't been updated since 2018.
* I don't encounter the problem I describe when sharing to EinkBro from any other source such as Feedly itself, Pocket or from any other Android browser to EinkBro.
* There's also no problem sharing a link/story from Palabre to any other browser (such as Chrome or Edge).

I'm doubtful that there's a setting of some kind I can tweak in EinkBro to properly enable sharing from Palabre to EinkBro, and supect like I say that this may be an issue with Palabre, but wanted to raise this here anyway in case someone can shed some light.

Again, congratuations on EinkBro. I look forward to it being my lightweight default browser when I get my eInk device.

Thank you.

**Comment by CottonEvoker** [comment]
I have a related question already, arising from this issue with Palabre, regarding sharing links from Feedly.

In light of the described issue with Palabre, I just reverted to Feedly, because sharing links from it to EinkBro seemed to work fine. 

However, I'm finding that if you share from Feedly to Einkbro, and then go 'back' in Einkbro (expecting to return to the articles overview in Feedly), this doesn't work: You remain in Einkbro without the ability to return to source app, Feedly in this instance. Back function appears to work differently in Einkbro than in Chrome/Edge. Depending on your selection in 'Behaviour' / 'Back key behaviour',  you can either have the page revert to the first/initial page opened in EinkBro, or have it close the tab. So instead of returning to Feedly, either the tab closes or else you get the message "No previous page".

Effectively, after sharing a link to Einkbro you cannot seem to then revert to the source app. (I find this to be the case with Einkbro also when shared to from Pocket, whilst (again) you can share from Pocket to Chrome or Edge, and then revert to Pocket via the back key or gesture.

This is actually rather inconveniencing/restrictive. Am I overlooking something about back functionality in EinkBro?

Thanks.

**Comment by plateaukao** [comment]
Hi @CottonEvoker 
For the first issue, please try to share the link to other apps (not browsers), for example sms app, messenger, or Google Keep, etc. So that I could know what exactly the share link is.
In EinkBor, if the shared content from other apps (in this case, Palabre) does not starts with http, then it will be treated as something that needs to be searched in search engine. So it's my guess that Palabre does not share a real link string.

For your second question. The behavior is due to how EinkBro handles links from other apps. To kind of solving this issue/behavior, you can configure the toolbar, to add a "minimize" button. It will hide EinkBro app, so that previous app would be displayed. I know it's not ideal, but that's the current way to make implementation of EinkBro as simple as possible.

**Reply by CottonEvoker** [reply]
Thank you so much for getting back.

**FIRST ISSUE**:

Okay, so as an example, let's take a news story from Wired: "How to Charge Your Phone Faster With Adaptive Charging". The article's url is "https://www.wired.com/story/how-to-charge-phone-faster-adaptive-charging".

**_In Palabre_**:

If I share this as a task in Palabre, say, to Microsoft To Do or to Todoist, in both cases it shares in a more or less identical way:

The task as created in MS To Do is named "How to Charge Your Phone Faster With Adaptive Charging.
https://www.wired.com/story/how-to-charge-phone-faster-adaptive-charging"

The task created in Todoist is named "How to Charge Your Phone Faster With Adaptive Charging. - https://www.wired.com/story/how-to-charge-phone-faster-adaptive-charging/"

The only difference between how MS To Do and Todoist handle the link seems to be the dash ("-") in the Todoist task between link title and link url.

So when the thus created task is viewed in either MS To Do or Todoist, link title and link url both show in the task name, meaning that you click on the task name (or on the url in the task name) to visit/open the link.

**_In Feedly_**:

If I share the very same link again as a task in Feedly, Microsoft To Do & Todoist each handles it slightly differently:

The task as created in MS To Do just contains the link title, i.e. task is named "How to Charge Your Phone Faster With Adaptive Charging". The url shows not in the task name, but as a task note.

Todoist's handling seems to be smarter/cleverer: The link as shared reads "[How to Charge Your Phone Faster With Adaptive Charging](https://www.wired.com/story/how-to-charge-phone-faster-adaptive-charging/)" [**] This is Todoist-specific formatting, and means that in Todoist, the task name (as per MS To Do), also reads "How to Charge Your Phone Faster With Adaptive Charging", but in Todoist the entire task name is itself a hyperlink, with the task name showing underlined to indicate this, i.e. that it's a link.

**SECOND ISSUE**:

This minimize button workaround seems to work for me. (I knew I was overlooking something!) Thank you!

[**] EDIT - okay, I just spotted that because of the way things are formatted here on Github, the above link share from Feedly to Todoist doesn't actually appear here exactly as it does whilst you're creating the shared task, but rather **precisely as it does once the task is created in Todoist**, i.e. with link title "How to Charge Your Phone Faster With Adaptive Charging" doubling as a url hyperlink. (I guess the link appearing this way here, and in Todoist, is markup, right, but I'm not that knowledgeable about these things?). Anyway, for completeness, [here's a screenshot](https://drive.google.com/file/d/1F1LQCxCEz8czH3SdaN9zH6xqjDWd41Hi/view?usp=sharing) of how the link shared from Feedly appears in the 'add task' box before being added to Todoist.

Another EDIT for further clarity, and given how link formatting on this post already distorted how at least one of the share links I pasted gets diplayed here. So here are screenshots for each of the share links created, as referenced for FIRST ISSUE above:

[Palabre link shared to Microsoft To Do](https://drive.google.com/file/d/13AWVme1t8YEoCrftoqFrL_VlvTUEJ4qw/view?usp=sharing)
[Palabre link shared to Todoist](https://drive.google.com/file/d/1Bx5vsYnPiSHRTrDuK3h142POpi7XNYhJ/view?usp=sharing)
[Feedly link shared to Microsoft To Do](https://drive.google.com/file/d/1iT4uba5i9307BOcOeL40V9sXMBODWaRJ/view?usp=sharing) (The url shows as a separate note within the task in To Do)
[Feedly link shared to Todoist](https://drive.google.com/file/d/1wRYnzU-8cHL-Vfhan0BNIWV1EcczSZqd/view?usp=sharing)

**Comment by plateaukao** [comment, marked as answer]
@CottonEvoker 
You could try enable Settings > Behaviour > Trim input text content. 

The reason it happens is that Palabre does not just share url. It also contains title in the content. So, its not recognized as a valid url in EinkBro.

**Reply by CottonEvoker** [reply]
That fixed it. Thank you so much Daniel, you're amazing.

Again I cannot praise eInkBro highly enough. That you have created this little browser that does so much more than the Titans of tech have been able to manage in their browsers is a remarkable feat that you should be very proud of.

Please send me a link to donate to you. I cannot find a donate link on any of your various pages.

EDIT: added "in their browsers"

**Reply by plateaukao** [reply]
@CottonEvoker 
Glad to know that EinkBro is helpful to you. I don't have any donation links for EinkBro.  I don't want to get any money from EinkBro itself.
If you really want to show your gratitude, you can consider buy [my book of describing how EinkBro is implemented](https://play.google.com/store/books/details/%E9%AB%98%E8%8C%82%E5%8E%9F_Daniel_Kao_Android%E9%96%8B%E6%BA%90%E5%B0%88%E6%A1%88_%E7%9C%9F_%E5%AF%A6%E6%88%B0%E5%95%9F%E8%88%AA?id=aOniEAAAQBAJ&hl=en&gl=US). It's written in Chinese though. :)

**Reply by CottonEvoker** [reply]
I was happy to purchase your book.
Now all I need to do is learn Chinese so I can read it!
Thank you again. 😀
